src/Model/Item.php

```diff
@@ -290,47 +288,47 @@ public static function insert($arr, $force_parent = false, $notify = false, $don
 
 		self::addLanguageInPostopts($arr);
 
-		$arr['wall']          = ((x($arr, 'wall'))          ? intval($arr['wall'])                : 0);
-		$arr['extid']         = ((x($arr, 'extid'))         ? notags(trim($arr['extid']))         : '');
-		$arr['author-name']   = ((x($arr, 'author-name'))   ? trim($arr['author-name'])   : '');
-		$arr['author-link']   = ((x($arr, 'author-link'))   ? notags(trim($arr['author-link']))   : '');
-		$arr['author-avatar'] = ((x($arr, 'author-avatar')) ? notags(trim($arr['author-avatar'])) : '');
-		$arr['owner-name']    = ((x($arr, 'owner-name'))    ? trim($arr['owner-name'])    : '');
-		$arr['owner-link']    = ((x($arr, 'owner-link'))    ? notags(trim($arr['owner-link']))    : '');
-		$arr['owner-avatar']  = ((x($arr, 'owner-avatar'))  ? notags(trim($arr['owner-avatar']))  : '');
+		$arr['wall']          = intval(defaults($arr, 'wall', 0));
+		$arr['extid']         = trim(defaults($arr, 'extid', ''));
+		$arr['author-name']   = trim(defaults($arr, 'author-name', ''));
+		$arr['author-link']   = trim(defaults($arr, 'author-link', ''));
+		$arr['author-avatar'] = trim(defaults($arr, 'author-avatar', ''));
+		$arr['owner-name']    = trim(defaults($arr, 'owner-name', ''));
+		$arr['owner-link']    = trim(defaults($arr, 'owner-link', ''));
+		$arr['owner-avatar']  = trim(defaults($arr, 'owner-avatar', ''));
```

Comment:
What about the `notags()` that was performed before?

Reply:
These function destroyed more than they helped, since they simply replaced "<" and ">" with "[" and "]". AFAIK we do have some other checks directly at the output.